### ISSUE TITLE
Button groups don't work with packed scenes

### ISSUE THREAD

**Issue by ddel-rio** [godotengine/godot#93559]
### Tested versions

The bug can be reproduced in both the last commit of the master branch (v4.3.beta.custom_build [e78dc83ee]) and in the latest official version of the engine (Godot v4.2.2.stable).

### System information

Windows 11 - NVIDIA GeForce RTX 4070 Laptop - Godot v4.2.2.stable - Vulkan(Forward+)

### Issue description

Buttons that are a packed scenes do not have the expected behavior if I convert them to toggles and add them to a button group.
By pressing them they can be activated but never deactivated.

### Steps to reproduce

1. Converts a button to a packed scene.
2. Add two or more of those buttons to a scene.
3. Enable the toggle mode property.
4. Have them share a button group resource.

### Minimal reproduction project (MRP)

[mrp.zip](https://github.com/user-attachments/files/15958686/mrp.zip)

**Comment by AThousandShips** [comment]
> and in the latest official version of the engine (Godot v4.2.1.stable).

The latest official build is 4.2.2, but if it occurs in 4.3 it would likely occur there too

**Comment by ddel-rio** [comment]
True, sorry, I tried it in 4.2.2 but I made a mistake when writing the number, I'm going to correct it.

**Comment by ddel-rio** [comment]
New info: setting the button group in the packed scene instead of externally doesn't work either.

**Comment by sancho2** [comment]
You have to select allow_unpress.   
Click on the button group, select edit and you will find that option there.

**Comment by ddel-rio** [comment]
> You have to select allow_unpress.

I've already tried.
Doing that just makes it work like a normal toggle, but doesn't give me the group benefit of not being able to enable more than one button at a time.

**Comment by sancho2** [comment]
I see what you mean now.    
I can confirm that when you make a button a scene and instantiate it, it no longer respects the button group or even a new button group and it will not unselect a button when another in the group is pressed.      

GODOT v4.2.stable.official [46dc27791]

**Comment by lnewmanjohnson18** [comment]
I am experiencing this behavior in current 4.3 stable

4.3 MRP
[MRP.zip](https://github.com/user-attachments/files/17800857/MRP.zip)

**Comment by Janglee123** [comment]
> I am experiencing this behavior in current 4.3 stable
> 
> 4.3 MRP [MRP.zip](https://github.com/user-attachments/files/17800857/MRP.zip)

I tested MRP.zip project on 4.4.1. 

The project has two problems, and solving them fixes the issue.

1) The button group resource has local to scene checked. 
Unchecking local to scene fixes the problem.  All 4 buttons worked together as expected.

2) Instanced buttons are overwritten with different button groups!
Making sure both buttons have the same button group solve the issue. Buttons worked together in both groups, but not across groups as expected.

The issue is either fixed on 4.4.1 or above problems were causing it.

**Comment by HolonProduction** [comment]
Closing as per the above comment
